ghpages.mk

```diff
@@ -41,6 +41,50 @@ else
 	@echo '</html>' >>$@
 endif
 
+branches.html:
+	@echo '<!DOCTYPE html>' >$@
+	@echo '<html>' >>$@
+	@echo '<head><title>$(GITHUB_REPO) branches</title>' >> $@
+	@echo '<style>li:target { background-color: lightgrey; }</style>' >>$@
+	@echo '</head>' >>$@
+	@echo '<body><ul>' >>$@
+	for BRANCH in `git branch | cut -b3-`; do \
+	  [ x"$${BRANCH}" = x"gh-issues" ] || [ x"$${BRANCH}" == x"gh-pages" ] && continue; \
+	  [ x"$${BRANCH}" = x"master" ] && BRANCH_LINK="./" || BRANCH_LINK="$${BRANCH}/" ;\
+	    echo '<li id="branch-'"$${BRANCH}"'">'"$${BRANCH} branch<ul>" >> $@; \
+	    for DRAFT in $(drafts); do \
+	      echo '<li>' >>@; \
+	      [ 1 -eq $(words $(drafts)) ] || echo "$${DRAFT}: " >>$@; \
+	      echo '<a href="'"$${BRANCH_LINK}$${DRAFT}"'.html" class="html">HTML version</a>, <a href="'"$${BRANCH_LINK}$${DRAFT}"'.txt">plain text</a>, <a href="'"https://tools.ietf.org/rfcdiff?url1=https://tools.ietf.org/id/$${DRAFT}.txt&url2=https://${GITHUB_USER}.github.io/${GITHUB_REPO}/$${BRANCH_LINK}$${DRAFT}.txt"'" class="diff">Compare to published draft</a></li>' >>$@; \
+	    done ; \
+	    echo '</ul></li>' >> $@; \
+	done
+	@echo '</ul>' >>$@
+	@echo '<script>/* @licstart  The following is the entire license notice for the' \
```

Comment:
This might be better moved into a file, just to maintain readability.  Then you can use cat $(LIBDIR)/template/branch.js here instead.